utils/biospecimen.js

```diff
@@ -212,12 +212,11 @@ const biospecimenAPIs = async (req, res) => {
             return res.status(405).json(getResponseJSON('Only GET requests are accepted!', 405));
         }
         if(req.query.masterSpecimenId) {
-            const masterSpecimenId = req.query.masterSpecimenId;
-            if(!masterSpecimenId) return res.status(400).json(getResponseJSON('Bad request!', 400));
             const { searchSpecimen } = require('./firestore');
-            const allSitesFlag = (req.query.allSitesFlag) ? true : false
+            const masterSpecimenId = req.query.masterSpecimenId;
```

Comment:
NIT: We could write `const { masterSpecimenID } = req.query` instead

Reply:
`const masterSpecimenId = req.query.masterSpecimenId;`
Above approach is a direct assignment. Your suggested way is useful only if we have multiple properties from an object.
@we-ai do you think any one of them has better performance?

Reply:
Direct assignment is slightly faster, but the performance differences are minimal. I tested on a small object: `const obj ={ a: { ab: { abc: 1 } }, b: 2, c: 3, d: 4, e: 5 }`. Tested on 1 billion operations on `const {abc} = obj.a.ab` and `const abc = obj.a.ab.abc` . Below are 10 test runs if you're interested:
```
[
  [ 'Data Destruction: 588ms', 'Direct Assignment: 564ms' ],
  [ 'Data Destruction: 609ms', 'Direct Assignment: 584ms' ],
  [ 'Data Destruction: 570ms', 'Direct Assignment: 521ms' ],
  [ 'Data Destruction: 555ms', 'Direct Assignment: 544ms' ],
  [ 'Data Destruction: 516ms', 'Direct Assignment: 498ms' ],
  [ 'Data Destruction: 509ms', 'Direct Assignment: 494ms' ],
  [ 'Data Destruction: 505ms', 'Direct Assignment: 489ms' ],
  [ 'Data Destruction: 496ms', 'Direct Assignment: 494ms' ],
  [ 'Data Destruction: 481ms', 'Direct Assignment: 500ms' ],
  [ 'Data Destruction: 479ms', 'Direct Assignment: 481ms' ]
]
```

Reply:
Thanks @we-ai